doc/specs/#1564 - Settings UI/spec.md

```diff
@@ -0,0 +1,163 @@
+
+We will also be adding a TerminalControl inside the settings UI to preview what the changes will look like before actually saving them to the settings.json file.
+
+#### We also considered: automatically save settings
+
+As users edit fields in the settings UI, they are automatically saved and written to the JSON file. This allows the user to see their settings changes taking place in real time.
+
+## UI/UX Design
+
+Layout of all of the settings per page can be found in [Appendix 1](#appendix-1).
+
+### Top-level navigation: more descriptive navigation
+
+The navigation menu is broken up into more digestible sections. This aligns more closely to other terminals. The following are the proposed navigation items:
+
+- General
+    - Launch
+    - Interaction
+    - Rendering
+- Appearance
+    - Global
+    - Color schemes
+    - Themes*
+- Profiles
+    - Defaults
+    - Enumerate profiles
+    - Add new
+- Keyboard
+- Mouse*
+- Command Palette*
+- Marketplace*
+
+\* Themes, mouse, command palette, and marketplace will be added once they're implemented.
+
+![Settings UI navigation 2](./navigation-2.png)
+
+#### We also considered: align with JSON
+
+The settings UI could have top-level navigation that aligns with the overall structure of the settings.json file. The following are the proposed navigation items:
+
+- Globals
+- Profiles
+- Color schemes
+- Bindings
+
+For Bindings, it would have key bindings, mouse bindings, and command palette inside it.
+
+![Settings UI navigation 1](./navigation.png)
+
+## Capabilities
+
+### Accessibility
+
+This will have to undergo full accessibility testing because it is a new UI element. All items inside the settings UI should be accessible by a screen reader and the keyboard. Additionally, all of the settings UI will have to be localized.
+
+### Security
+
+This does not impact security.
+
+### Reliability
+
+This will not improve reliability.
+
+### Compatibility
+
+This will change the default experience to open the UI, rather than the JSON file in a text editor. This behavior can be reverted with the setting listed [above](#ability-to-disable-displaying-the-settings-ui).
+
+### Performance, Power, and Efficiency
+
+This does not affect performance, power, nor efficiency.
+
+## Potential Issues
+
+## Future considerations
+
+- We will have to have design reviews for all of the content pages.
+- We should have undo functionality. In a text editor, you can type `Ctrl+Z` however the settings UI is a bit more complex.
+- Once we have a marketplace for themes and extensions, this should be added to the top-level navigation.
+- As we add more features, the top-level navigation is subject to change in favor of improved usability.
+
+## Resources
+
+## Appendix 1
+
+Below is the list of all settings on their respective pages in the settings UI. The title row aligns with the navigation view on the left of the UI. Bolded headers in those columns align with top nav on the page.

```

Comment:
Anything with a path should also have a "browse" option:
- Background image
- Command line
- Icon
- Starting directory

[Here's](https://docs.microsoft.com/en-us/windows/uwp/files/quickstart-using-file-and-folder-pickers) a doc that might help with that?